app/code/Magento/Backend/view/adminhtml/web/js/store-switcher.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery'
+], function ($) {
+    'use strict';
+
+    return function (config) {
+        var scopeSwitcherHandler;
+
+        (function () {
+            var storesList = $('[data-role=stores-list]');
+
+            storesList.on('click', '[data-value]', function (event) {
+                var val = $(event.target).data('value'),
+                    role = $(event.target).data('role'),
+                    switcher = $('[data-role=' + role + ']');
+
+                event.preventDefault();
+
+                if (!switcher.val() || val !== switcher.val()) {
+
+                    /* Set the value & trigger event */
+                    switcher.val(val).trigger('change');
+                }
+            });
+        })($);
+
+        /**
+         * Switch store scope
+         *
+         * @param {Object} obj
+         * @return void
+         */
+        function switchScope(obj) {
+            var switcher = $(obj),
+                scopeId = switcher.val(),
+                scopeParams = '',
+                switcherParams = {};
+
+            if (scopeId) {
+                scopeParams = switcher.data('param') + '/' + scopeId + '/';
+            }
+
+            if (obj.switchParams) {
+                scopeParams += obj.switchParams;
+            }
+
+            /**
+             * Reload function for switcher
+             */
+            function reload() {
+                var url;
+
+                if (!config.isUsingIframe) {
+
+                    if (config.switchUrl && config.switchUrl.length > 0) {
+                        url = config.switchUrl + scopeParams;
+
+                        /* eslint-disable no-undef */
+                        setLocation(url);
+                    }
+
+                } else {
+                    $('#preview_selected_store').val(scopeId);
+                    $('#preview_form').submit();
+
+                    $('.store-switcher .dropdown-menu li a').each(function () {
+                        var $this = $(this);
+
+                        if ($this.data('role') === 'store-view-id' && $this.data('value') === scopeId) {
+                            $('#store-change-button').html($this.text());
+                        }
+                    });
+
+                    $('#store-change-button').click();
+                }
+            }
+
+            if (typeof scopeSwitcherHandler !== 'undefined') {
+                switcherParams = {
+                    scopeId: scopeId,
+                    scopeParams: scopeParams,
+                    useConfirm: config.useConfirm
+                };
+
+                scopeSwitcherHandler(switcherParams);
+            } else if (config.useConfirm) {
+                require([
+                    'mage/translate',
+                    'Magento_Ui/js/modal/confirm'
+                ], function ($t, confirm) {
```

Comment:
I believe moving to a separate require section broke functional tests, as this confirmation popup appears not immediately anymore.
There two options:
1. fix the failing action groups
2. move back these dependencies to the global `define` block of this file.

Actually, I'm fine with both options.

Reply:
i have updated follow option 2 for minimal function break and less impact to performance when backend we don't need to serious about performance